internal/services/pam/pam_test.go

```diff
@@ -815,40 +815,55 @@ func startSession(t *testing.T, client authd.PAMClient, username string) string
 	return sbResp.GetSessionId()
 }
 
-func TestMain(m *testing.M) {
-	if os.Getenv("GO_WANT_HELPER_PROCESS") != "" {
-		os.Exit(m.Run())
-	}
-
-	slog.SetLogLoggerLevel(slog.LevelDebug)
+// setupGlobalBrokerMock creates and points to a test-wide system bus, registering the mock broker on it.
+func setupGlobalBrokerMock() (cleanup func(), err error) {
+	cleanup = func() {}
 
 	// Start system bus mock.
 	busCleanup, err := testutils.StartSystemBusMock()
 	if err != nil {
-		fmt.Fprintf(os.Stderr, "%v\n", err)
-		os.Exit(1)
+		return cleanup, err
 	}
-	defer busCleanup()
+	cleanup = busCleanup
 
 	// Start brokers mock over dbus.
-	brokersConfPath, cleanup, err := initBrokers()
+	brokersConfPath, brokerCleanup, err := initBrokers()
 	if err != nil {
-		fmt.Fprintf(os.Stderr, "%v\n", err)
-		os.Exit(1)
+		return cleanup, err
+	}
+
+	cleanup = func() {
+		brokerCleanup()
+		busCleanup()
 	}
-	defer cleanup()
 
 	// Get manager shared across grpc services.
 	globalBrokerManager, err = brokers.NewManager(context.Background(), brokersConfPath, nil)
 	if err != nil {
-		fmt.Fprintf(os.Stderr, "%v\n", err)
-		os.Exit(1)
+		return cleanup, err
 	}
 	mockBrokerGeneratedID, err = getMockBrokerGeneratedID(globalBrokerManager)
 	if err != nil {
+		return cleanup, err
+	}
+
+	return cleanup, nil
+}
+
+func TestMain(m *testing.M) {
+	if os.Getenv("GO_WANT_HELPER_PROCESS") != "" {
+		os.Exit(m.Run())
```

Comment:
Not an issue here, but maybe it's better to split this in two to make it clearer it's not the best practice?
So:

```suggestion
		m.Run()
		os.Exit(0)
```

Reply:
good idea! Will do that!

Reply:
or double thinking about that, we should maybe panic? I think we can expect the subprocess to exit always and this would be a way for us to ensure that. As you approved, I’m adding that in the back of my mind and we can chat :)

Reply:
Ah, that's indeed even better, if that's the expectation. As at least it works as a _comment_.